src/dmd/dstruct.d

```diff
@@ -479,6 +479,19 @@ extern (C++) class StructDeclaration : AggregateDeclaration
             Type origType = t;
             Type tb = t.toBasetype();
 
+            const hasPointers = tb.hasPointers();
```

Comment:
seems to me like this block would be big enough to put it into a function.
This would help preventing diverging error messages.